### DISCUSSION TITLE
git_commit发生变化时config.h,in生成的头文件中没有更新

### DISCUSSION THREAD

**Discussion by TinyMiao** [xmake-io/xmake#4394] [Q&A] [answered]
config.h,.in内容如下
`
#define BEZIER_BUILD_TIME "${VERSION_BUILD}"
#define BEZIER_GIT_COMMIT "${GIT_COMMIT}"
`
当git_commit改变时，从新编译时，编译时间和commit对应的宏没有更新，依然还是上次的值
但是当删除掉生成的config.h后重新编译将会更新，请问有没有更好的解决方案呢？

**Comment by waruqi** [comment, marked as answer]
你要执行 `xmake f -c` 去强制重新检测配置

**Reply by TinyMiao** [reply]
请问`xmake f -c`是代表`xmake f --clean`还是`xmake f --check`

**Reply by waruqi** [reply]
clean，清除本地config cache 会强制检测，--check仅仅检测，但不会清所有cache

**Reply by TinyMiao** [reply]
每次编译完后删除生成的.h是不是也可以达到这样的目的呢
```lua
    after_build(function (target)
        if os.isdir("$(buildir)/config") then
            os.rm("$(buildir)/config")
            print("rm $(buildir)/config")
        end
    end)
```

**Reply by waruqi** [reply]
这会导致 config.h 的 modify time 总是会变，依赖它的文件都会每次重新编译。。
